CefSharp.Core/ManagedCefBrowserAdapter.cpp

```diff
@@ -0,0 +1,32 @@
+#include "Stdafx.h"
+#include "Cef.h"
+
+double ManagedCefBrowserAdapter::GetZoomLevel()
+{
+    auto browser = _clientAdapter->GetCefBrowser();
+
+    if (browser != nullptr)
+    {
+        auto host = browser->GetHost();
+        if (CefCurrentlyOn(TID_UI))
+        {
+            return host->GetZoomLevel();
+        }
+        else
+        {
+            // TODO: Add an async version of GetZoomLevel at some point.
+            // NOTE: Use of ManualResetEvent is required here in order
+            // for simple marshaling of some kind of synchronization primitive
+            // to the callback.
+            Task<double>^ task = Cef::UIThreadTaskFactory->StartNew(gcnew Func<double>(this, &ManagedCefBrowserAdapter::GetZoomLevel));
```

Comment:
We can totally remove the sync version as a breaking change. We'd need to either export `CefCurrentlyOn` or implement `SyncronizationContexts` in order to have the non-async version in C#. I'd rather not do `SyncronizationContexts` since I think ensuring the current context got updated correctly wouldn't be maintainable long term. (since you'd have to call the context installation method on every callback from CEF to .Net)

No argument on the change you suggest to `IWebBrowser`.